src/DataViews/SamplingReportDataView.cpp

```diff
@@ -315,10 +331,14 @@ void SamplingReportDataView::SetSampledFunctions(const std::vector<SampledFuncti
 
 void SamplingReportDataView::SetThreadID(ThreadID tid) {
   tid_ = tid;
+  const orbit_client_data::CaptureData& capture_data = app_->GetCaptureData();
+
   if (tid == orbit_base::kAllProcessThreadsTid) {
-    name_ = absl::StrFormat("%s\n(all threads)", app_->GetCaptureData().process_name());
+    name_ = absl::StrFormat("%s\n(all threads)", capture_data.process_name());
+    thread_events_count_ = capture_data.GetCallstackData().GetCallstackEventsCount();
   } else {
-    name_ = absl::StrFormat("%s\n[%d]", app_->GetCaptureData().GetThreadName(tid_), tid_);
+    name_ = absl::StrFormat("%s\n[%d]", capture_data.GetThreadName(tid_), tid_);
+    thread_events_count_ = capture_data.GetCallstackData().GetCallstackEventsOfTidCount(tid_);
```

Comment:
What happens here during capture?

Reply:
Also, does that work with selections?

Reply:
Great point about selections, thank you! But what might happen during capture? The `Sampling` and `Sampling (selection)` tabs are not active during the capture.

Reply:
As far as I remember it is possible to do selections during a capture. The overall "sampling" tab will not get populated, though.

Reply:
So yeah, back to my other comment, I think that the total count should be passed in from `ThreadSampleData::samples_count`, the same way `std::vector<SampledFunction>` is passed in. The thread name case is probably for convenience, but the pattern here is that data is set from the outside rather than being retrieved from `app_->GetCaptureData();`.